drivers/spi/spi_ll_stm32.c

```diff
@@ -0,0 +1,442 @@
+
+	for (br = 1 ; br <= ARRAY_SIZE(scaler) ; ++br) {
+		u32_t clk = clock >> br;
+
+		if (clk < config->frequency) {
+			break;
+		}
+	}
+
+	if (br > ARRAY_SIZE(scaler)) {
+		return -ENOTSUP;
+	}
+
+	LL_SPI_Disable(spi);
+	LL_SPI_SetBaudRatePrescaler(spi, scaler[br - 1]);
+
+	if (SPI_MODE_GET(config->operation) ==  SPI_MODE_CPOL) {
+		LL_SPI_SetClockPolarity(spi, LL_SPI_POLARITY_HIGH);
+	} else {
+		LL_SPI_SetClockPolarity(spi, LL_SPI_POLARITY_LOW);
+	}
+
+	if (SPI_MODE_GET(config->operation) == SPI_MODE_CPHA) {
+		LL_SPI_SetClockPhase(spi, LL_SPI_PHASE_2EDGE);
+	} else {
+		LL_SPI_SetClockPhase(spi, LL_SPI_PHASE_1EDGE);
+	}
+
+	LL_SPI_SetTransferDirection(spi, LL_SPI_FULL_DUPLEX);
+
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		LL_SPI_SetTransferBitOrder(spi, LL_SPI_LSB_FIRST);
+	} else {
+		LL_SPI_SetTransferBitOrder(spi, LL_SPI_MSB_FIRST);
+	}
+
+	LL_SPI_DisableCRC(spi);
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		LL_SPI_SetMode(spi, LL_SPI_MODE_SLAVE);
+	} else {
+		LL_SPI_SetMode(spi, LL_SPI_MODE_MASTER);
+	}
+
+	if (config->vendor & STM32_SPI_NSS_IGNORE) {
+		LL_SPI_SetNSSMode(spi, LL_SPI_NSS_SOFT);
+	} else {
+		if (config->operation & SPI_OP_MODE_SLAVE) {
+			LL_SPI_SetNSSMode(spi, LL_SPI_NSS_HARD_OUTPUT);
+		} else {
+			LL_SPI_SetNSSMode(spi, LL_SPI_NSS_SOFT);
+		}
+	}
+
+	LL_SPI_SetDataWidth(spi, LL_SPI_DATAWIDTH_8BIT);
+
+#if defined(CONFIG_SOC_SERIES_STM32L4X) || defined(CONFIG_SOC_SERIES_STM32F3X)
+	LL_SPI_SetRxFIFOThreshold(spi, LL_SPI_RX_FIFO_TH_QUARTER);
+#endif
+	LL_SPI_SetStandard(spi, LL_SPI_PROTOCOL_MOTOROLA);
+
+	/* At this point, it's mandatory to set this on the context! */
+	data->ctx.config = config;
+
+	spi_context_cs_configure(&data->ctx);
+
+	return 0;
+}
+
+static void spi_stm32_transmit(SPI_TypeDef *spi, struct spi_stm32_data *data)
+{
+	if (spi_context_tx_on(&data->ctx)) {
+		LL_SPI_TransmitData8(spi, UNALIGNED_GET((u8_t *)
+							(data->ctx.tx_buf)));
+	} else {
+		/* Transmit NOP byte */
+		LL_SPI_TransmitData8(spi, 0);
+	}
+
+	spi_context_update_tx(&data->ctx, 1);
+}
+
+static void spi_stm32_receive(SPI_TypeDef *spi, struct spi_stm32_data *data)
+{
+	if (spi_context_rx_on(&data->ctx)) {
+		u8_t byte = LL_SPI_ReceiveData8(spi);
+
+		UNALIGNED_PUT(byte, (u8_t *)data->ctx.rx_buf);
+	} else {
+		LL_SPI_ReceiveData8(spi);
+	}
+
+	spi_context_update_rx(&data->ctx, 1);
+}
+
+static int spi_stm32_release(struct spi_config *config)
+{
+	struct spi_stm32_data *data = CONFIG_DATA(config);
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static int transceive(struct spi_config *config,
+		      const struct spi_buf *tx_bufs, u32_t tx_count,
+		      struct spi_buf *rx_bufs, u32_t rx_count,
+		      bool asynchronous, struct k_poll_signal *signal)
+{
+	const struct spi_stm32_config *cfg = CONFIG_CFG(config);
+	struct spi_stm32_data *data = CONFIG_DATA(config);
+	SPI_TypeDef *spi = cfg->spi;
+	int ret;
+
+	if (!tx_count && !rx_count)
```

Comment:
don't forget the {} even for one liner statements on if/for/while. Apply this everywhere.